CAIPs/caip-282.md

```diff
@@ -0,0 +1,183 @@
+
+## Test Cases
+
+Here are some illustrative examples for both JSON-RPC request params:
+
+```typescript
+// Example for wallet_prompt
+{
+  id: 1,
+  jsonrpc: "2.0"
+  method: "wallet_prompt",
+  params: {
+    chains: ["chain:777", "chain:888", "chain:999"]  // optional
+    //  if the Blockchain Library supports CAIP-275 then it can include a name
+    authName: "johndoe.chain", // optional
+  }
+}
+
+
+// Example for wallet_announce
+{
+  id: 2,
+  jsonrpc: "2.0"
+  method: "wallet_announce",
+  params: {
+    uuid:  "350670db-19fa-4704-a166-e52e178b59d2",
+    name: "Example Wallet",
+    icon: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==",
+    rdns: "com.example.wallet",
+  }
+}
+```
+
+## Security Considerations
+
+Regarding security considerations it's important to consider that the WalletData can be imitated and/or manipulated by cross-site scripting and additionally we must consider the following:
+
+### Wallet Imitation and Manipulation
+
+Application developers are expected to actively detect for misbehavior of properties or functions being modified in order to tamper with or modify other wallets. One way this can be easily achieved is to look for when the uuid property within two WalletData objects match. Applications and Libraries are expected to consider other potential methods that the WalletData objects are being tampered with and consider additional mitigation techniques to prevent this as well in order to protect the user.
+
+### Prevent SVG Javascript Execution
+
+The use of SVG images introduces a cross-site scripting risk as they can include JavaScript code. This JavaScript executes within the context of the page and can modify the page or the contents of the page. So, when considering the experience of rendering the icons, dapps need to take into consideration how they’ll approach handling these concerns in order to prevent an image from being used as an obfuscation technique to hide malicious modifications to the page or to other wallets.
+
+## Privacy Considerations
+
+Any form of wallet discoverability must alwasys take in consideration wallet fingerprinting that can happen by malicious webpages or extensions that attempt to capture user information. Wallet Providers can abstain from publishing "Announce" messages on every page load and wait for incoming "Prompt" messages. Yet this opens the possibility for race conditions where Wallet Providers could be initialized after the "Prompt" message was published and therefore be undiscoverable. It is recommended that Wallet Providers offer this more "private connect" feature that users only enable optionally, rather than set by default.
```

Comment:
```suggestion
Any form of wallet discoverability must always take into consideration wallet fingerprinting that can happen by malicious webpages or extensions that attempt to capture user information. Wallet Providers can abstain from publishing "Announce" messages on every page load and wait for incoming "Prompt" messages. Yet this opens the possibility for race conditions where Wallet Providers could be initialized after the "Prompt" message was published and therefore be undiscoverable. It is recommended that Wallet Providers offer this more "private connect" feature that users only enable optionally, rather than set by default.
```